camayoc/tests/qcs/test_credentials.py

```diff
@@ -185,11 +275,21 @@ def test_read_all(cleanup):
     :expectedresults: All hosts are present in data returned by API.
     :caseautomation: notautomated
     """
-    pass
+    host_credentials = []
+    for _ in range(random.randint(2, 5)):
+        cred = HostCredential(client=shared_client, password=uuid4())
+        cred.create()
+        # add the credential to the list to destroy after the test is done
+        cleanup.append(cred)
+        assert_matches_server(cred)
+        host_credentials.append(cred)
 
+    remote_host_credentials = HostCredential().list().json()
+    for local, remote in zip(host_credentials, remote_host_credentials):
+        local.equivalent(remote)
```

Comment:
It seems that this is dependent that the host credentials got created in the same order as we sent them? 
Do we want to require that?

Reply:
An option is to sort both lists by their id before zip, what do you think?

Reply:
I am not intimately familar with what zip does, but will it be happy if the two lists are not the same length? 

I thought of this in regards to the concerns about being "parallel safe" i.e. maybe there will be other activities on the server. 

Maybe what you do on line 315:
```diff
+    for host_credential in host_credentials:
+        remote = HostCredential(_id=host_credential._id).read().json()
+        host_credential.equivalent(remote)
```
would be better?

Reply:
This is an option but will do more GET requests, but I think it is fine since we are exercising the server. If that become an issue in the future we do another round of improvement.

Reply:
sounds good